### PR TITLE
Functional tests - Add first UI test of discount V2 : Minimum purchase amount (On cart amount)

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.1.x
| Description?      | Add first UI test of discount V2 : Minimum purchase amount (On cart amount)
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Ci and nightly are :green_apple: 
| UI Tests          | https://github.com/nesrineabdmouleh/ga.tests.ui.pr/actions/runs/22619364974
| Fixed issue or discussion?     | no
| Related PRs       | no
| Sponsor company   | @PrestaShopCore

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+    name: ' ',
+    noProductCondition: true,
+    minimumPurchaseAmount: true,
+    minimumAmountValue: 20,
+    minimumAmountTax: 'Tax included',
+    discountValue: 10,
+    discountReductionType: '€',
+    discountTax: 'Tax included',
+  });
+
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts]:
```suggestion
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+    name: 'Test',
+    noProductCondition: true,
+    minimumPurchaseAmount: true,
+    minimumAmountValue: 'asefr',
+    minimumAmountTax: 'Tax included',
+    discountValue: 10,
+    discountReductionType: '€',
+    discountTax: 'Tax included',
+  });
+
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts]:
```suggestion
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+    name: 'Test',
+    noProductCondition: true,
+    minimumPurchaseAmount: true,
+    minimumAmountValue: 50,
+    minimumAmountTax: 'Tax included',
+    discountValue: -20,
+    discountReductionType: '€',
+    discountTax: 'Tax included',
+  });
+
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts]:
```suggestion
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+    name: 'Test',
+    noProductCondition: true,
+    minimumPurchaseAmount: true,
+    minimumAmountValue: 50,
+    minimumAmountTax: 'Tax included',
+    discountValue: '0',
+    discountReductionType: '€',
+    discountTax: 'Tax included',
+  });
+
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts:110]:
```suggestion
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+        page,
+        boDashboardPage.catalogParentLink,
+        boDashboardPage.discountsLink,
+      );
+
+      const pageTitle = await boDiscountsPage.getPageTitle(page);
+      expect(pageTitle).to.contains(boDiscountsPage.pageTitle);
+    });
+
+    it('should click on create discount button and choose the type \'On cart amount\'', async function () {
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts]:
```suggestion
    it(`should click on create discount button and choose the type '${discountWithoutName.discountType!}'`, async function () {
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+      const pageTitle = await foHummingbirdCartPage.getPageTitle(page);
+      expect(pageTitle).to.contains(foHummingbirdCartPage.pageTitle);
+    });
+
+    it('should check the discount value', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkDiscountValue_2', baseContext);
+
+      await foHummingbirdCartPage.reloadPage(page);
+      const discountValue = await foHummingbirdCartPage.getCartRuleValue(page);
+      expect(discountValue).to.contains('-€12.00');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts]: Could you calculate it from dataProducts & editDiscountData ?

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts [hunk 7]
```diff
@@ -0,0 +1,448 @@
...
+
+      const subTotalShipping = await foHummingbirdCartPage.getSubtotalShippingValue(page);
+      expect(subTotalShipping).to.eq('Free');
+    });
+
+    it('should check the Total (tax incl.) after the discount', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkTotalAfterDiscount_2', baseContext);
+
+      const totalAfterDiscount = await foHummingbirdCartPage.getATIPrice(page);
+      expect(totalAfterDiscount.toString()).to.equal('57.26');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/discountV2/01_minimumPurchaseAmount.ts]: Could you calculate it from dataProducts & editDiscountData ?

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

**Review by Progi1984 — APPROVED** [review]
